lib/src/Base/Optim/Cobyla.cxx

```diff
@@ -149,10 +150,11 @@ void Cobyla::run()
     result_.store(inP, Point(1, outP[0]), absoluteError, relativeError, residualError, constraintError);
   }
 
-  result_.setOptimalPoint(x);
-  const UnsignedInteger index = evaluationInputHistory_.find(x);
-  Scalar bestValue = evaluationOutputHistory_(index, 0);
-  result_.setOptimalValue(Point(1, bestValue));
+  // x might not be exactly the best point evaluated, so find the nearest
+  const UnsignedInteger optimalIndex = NearestNeighbourAlgorithm(evaluationInputHistory_).query(x);
```

Comment:
But building any NearestNeighbourAlgorithm is costly, I am pretty sure that it is not worth when querying a single point.
You should only do that if `evaluationInputHistory_.find(x)` fails.

Reply:
Ok, done